odxtools/parameters/parameterwithdop.py

```diff
@@ -39,20 +40,20 @@ def dop(self) -> Union[DopBase, None]:
     def dop(self, dop: DopBase):
         self._dop = dop
         if not self.dop_snref:
-            self.dop_ref = dop.id
+            self.dop_ref = OdxLinkRef.from_id(dop.id)
         else:
             self.dop_snref = dop.short_name
 
-    def resolve_references(self, parent_dl, id_lookup):
+    def resolve_references(self, parent_dl, odxlinks: OdxLinkDatabase):
         dop = self.dop
         if self.dop_snref:
-            dop = parent_dl.data_object_properties[self.dop_snref]
-            if not dop:
+            dop = parent_dl.data_object_properties.get(self.dop_snref)
+            if dop is None:
                 logger.info(
                     f"Param {self.short_name} could not resolve DOP-SNREF {self.dop_snref}")
         elif self.dop_ref:
-            dop = id_lookup.get(self.dop_ref)
-            if not dop:
+            dop = odxlinks.resolve_lenient(self.dop_ref) # TODO: non-lenient!
```

Comment:
Only used here, may try/except instead of double implementation?

Reply:
yes, the database is correct: it contains some static fields which are not yet internalized by odxtools but which are referenced. (I'll implement static fields soon.)